kubernetes/kubetest/mock.go

```diff
@@ -131,3 +149,243 @@ func (o *K8SClientMock) GetQuotaSpecBinding(namespace string, quotaSpecBindingNa
 	args := o.Called(namespace, quotaSpecBindingName)
 	return args.Get(0).(kubernetes.IstioObject), args.Error(1)
 }
+
+func (o *K8SClientMock) FakeServiceDetails() *kubernetes.ServiceDetails {
```

Comment:
Love it! I think we should place some more fake methods in order to re-use it from test. Good idea :)